### ISSUE TITLE
language_detection function may have some fault with upper case English.

### ISSUE THREAD

**Issue by svjack** [UKPLab/EasyNMT#2]
language_detection default use fasttext lid.176.ftz
which will have some confusion between upper case English and Japanese.
following is a example:

model = EasyNMT("opus-mt")
text_list = ["WHAT SCHOOL DOES HA'SEAN CLINTON-DIX BELONG TO?",
  'WHAT IS THE LOWEST ROUND FOR CB POSITION?',
  'IN THE ISSUE WHERE NICOLE NARAIN WAS THE CENTERFOLD MODEL, WHO WAS THE INTERVIEW SUBJECT?',
  "WHAT WAS ESSENDON'S HIGHEST CROWD NUMBER WHEN PLAYING AS THE AWAY TEAM?",
  'WHAT ARE THE HIGHEST POINTS WITH GOALS LARGER THAN 48, WINS LESS THAN 19, GOALS AGAINST SMALLER THAN 44, DRAWS LARGER THAN 5?',
  'WHAT IS THE STROKE COUNT WITH RADICAL OF 生, FRQUENCY SMALLER THAN 22?',
  'HOW MANY TEMPERATURE INTERVALS ARE POSSIBLE TO USE WITH ACRYL? ',
  'WHAT ARE THE GOALS WITH DRAWS SMALLER THAN 6, AND LOSSES SMALLER THAN 7?',
  'Name the least number for  君のそばで～ヒカリのテーマ～(popup.version)',
  'WHAT WAS THE SURFACE MADE OF WHEN THE OPPONENT WAS GEORGE KHRIKADZE?',
  'Name the english name for  福鼎市',
  'WHAT IS THE LOWEST MONEY WITH 74-74-76-74=298 SCORE?',
  'WHAT IS THE HIGHEST PICK WITH A TE POSITION, AND ROUND SMALLER THAN 2?',
  'WHAT IS THE LOWEST MONEY WITH TO PAR LARGER THAN 26?']

pd.Series(text_list).map(model.language_detection).value_counts(),
pd.Series(text_list).map(lambda x: x.lower()).map(model.language_detection).value_counts()

first series detect all to ja, second distinguish en and ja.
May have some pre-process of input , or change the default detector.

**Comment by nreimers** [comment]
Hi @svjack 
Thanks for spotting this.

Yes, pre-trained models are sadly often quite sensitive for casing. This is also a common issues for POS and NER: https://arxiv.org/abs/1903.11222

I think the issue is that fastText was trained only on clean data from  Wikipedia, Tatoeba and SETimes and has never seen noisy data.

I tested to lower case everything:
```python
from easynmt import EasyNMT
model = EasyNMT('opus-mt')
text_list = ["WHAT SCHOOL DOES HA'SEAN CLINTON-DIX BELONG TO?",
'WHAT IS THE LOWEST ROUND FOR CB POSITION?',
'IN THE ISSUE WHERE NICOLE NARAIN WAS THE CENTERFOLD MODEL, WHO WAS THE INTERVIEW SUBJECT?',
"WHAT WAS ESSENDON'S HIGHEST CROWD NUMBER WHEN PLAYING AS THE AWAY TEAM?",
'WHAT ARE THE HIGHEST POINTS WITH GOALS LARGER THAN 48, WINS LESS THAN 19, GOALS AGAINST SMALLER THAN 44, DRAWS LARGER THAN 5?',
'WHAT IS THE STROKE COUNT WITH RADICAL OF 生, FRQUENCY SMALLER THAN 22?',
'HOW MANY TEMPERATURE INTERVALS ARE POSSIBLE TO USE WITH ACRYL? ',
'WHAT ARE THE GOALS WITH DRAWS SMALLER THAN 6, AND LOSSES SMALLER THAN 7?',
'Name the least number for 君のそばで～ヒカリのテーマ～(popup.version)',
'WHAT WAS THE SURFACE MADE OF WHEN THE OPPONENT WAS GEORGE KHRIKADZE?',
'Name the english name for 福鼎市',
'WHAT IS THE LOWEST MONEY WITH 74-74-76-74=298 SCORE?',
'WHAT IS THE HIGHEST PICK WITH A TE POSITION, AND ROUND SMALLER THAN 2?',
'WHAT IS THE LOWEST MONEY WITH TO PAR LARGER THAN 26?']

for sent in text_list:
  print(sent)
  print("==> Language as is:", model.language_detection(sent))
  print("==> Language lower:", model.language_detection(sent.lower()))
``` 

and it worked much better for the presented sentences.

Will need to check if text.lower() will do the trick for fastText or if it will have negative consequences for other languages that use casing.

**Comment by nreimers** [comment]
Yes, it appears that fasttext language identification does not work well with UPPERCASE text.

Converting text to lower case before language identification gives much better results.

I released version 1.0.1 which includes this fix.
